### PR TITLE
Silence logging by default

### PR DESCRIPTION
**Pull request by akaIDIOT**
As applied by many libs and recommended by some [source](https://realpython.com/python-logging-source-code/#library-vs-application-logging-what-is-nullhandler)s, default all of the `confidence.*` logging handlers to be silent; logging should be an opt-in.

Small local test makes the logging show up when using `logging.basicConfig(level=logging.DEBUG)`, so not sure exactly how silent this makes the lib (doing this before importing the io module didn't change that).

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -14,13 +14,13 @@ pluggy==1.0.0
     # via pytest
 py==1.11.0
     # via pytest
-pyparsing==3.0.6
+pyparsing==3.0.7
     # via packaging
-pytest==6.2.5
+pytest==7.0.0
     # via -r test-requirements.in
-toml==0.10.2
```
Comment (bartbroere) [inline test-requirements.txt:21]: TIL

File: CHANGES.md [hunk 2]
```diff
@@ -4,6 +4,8 @@ Changes
 development (master)
 --------------------
 
+- Use named loggers, default `confidence.*` loggers to silence
```
Comment (bartbroere) [inline CHANGES.md:7]: From the diff I'm missing the reason for defaulting logging to silence.

**Review by bartbroere — APPROVED** [review]

File: CHANGES.md (hunk 2 above)
Comment (akaIDIOT) [reply CHANGES.md:7]: A good point, let's add something along the lines of the PR description to `CHANGES.md`. Maybe also mention this in the docs, even though those are rather glass-half-empty still.

- [ ] add rationale for silent loggers to changelog
- [ ] add logging section to docs

File: test-requirements.txt (hunk 1 above)
Comment (akaIDIOT) [reply test-requirements.txt:21]: Hmm curious, must have been a change between pytest 6 and 7, dont think this will mean anything for confidence though.

Comment (akaIDIOT) [reply test-requirements.txt:21]: No idea what the difference is, but decided I don't really care :)

**Pull request merged**